test/core/credentials/call/dual/dual_call_credentials_test.cc

```diff
@@ -0,0 +1,146 @@
+                GRPC_TRANSPORT_SECURITY_TYPE_PROPERTY_NAME,
+                GRPC_ALTS_TRANSPORT_SECURITY_TYPE)));
+    channel_creds_ = RefCountedPtr<grpc_composite_channel_credentials>(creds);
+    pollent_ =
+        grpc_polling_entity_create_from_pollset_set(grpc_pollset_set_create());
+  }
+
+  void TearDown() override {
+    grpc_pollset_set_destroy(grpc_polling_entity_pollset_set(&pollent_));
+  }
+
+  void RunRequestMetadataTest(
+      grpc_call_credentials::GetRequestMetadataArgs get_request_metadata_args) {
+    activity_ = MakeActivity(
+        [this, &get_request_metadata_args] {
+          return Map(channel_creds_->mutable_call_creds()->GetRequestMetadata(
+                         ClientMetadataHandle(&expected_md_,
+                                              Arena::PooledDeleter(nullptr)),
+                         &get_request_metadata_args),
+                     [](absl::StatusOr<ClientMetadataHandle> metadata) {
+                       return metadata.status();
+                     });
+        },
+        ExecCtxWakeupScheduler(),
+        [](absl::Status status) mutable { ASSERT_TRUE(status.ok()); },
+        arena_.get(), &pollent_);
+  }
+
+  RefCountedPtr<grpc_auth_context> CreateAuthContextWithSecurityType(
+      std::string security_type) {
+    auto auth_context = MakeRefCounted<grpc_auth_context>(nullptr);
+    grpc_auth_context_add_property(
+        auth_context.get(), GRPC_TRANSPORT_SECURITY_TYPE_PROPERTY_NAME,
+        security_type.c_str(), security_type.length());
+    grpc_auth_context_set_peer_identity_property_name(
+        auth_context.get(), GRPC_TRANSPORT_SECURITY_TYPE_PROPERTY_NAME);
+    return auth_context;
+  }
+
+  RefCountedPtr<Arena> arena_ = SimpleArenaAllocator()->MakeArena();
+  grpc_metadata_batch expected_md_;
+  grpc_polling_entity pollent_;
+  ActivityPtr activity_;
+  RefCountedPtr<grpc_composite_channel_credentials> channel_creds_;
+};
+
+TEST_F(DualCredentialsTest, UseAltsCredentials) {
+  ExecCtx exec_ctx;
+  grpc_call_credentials::GetRequestMetadataArgs get_request_metadata_args = {
+      nullptr,
+      CreateAuthContextWithSecurityType(GRPC_ALTS_TRANSPORT_SECURITY_TYPE)};
+
+  RunRequestMetadataTest(get_request_metadata_args);
+
+  std::string buffer;
+  EXPECT_EQ(expected_md_.GetStringValue(
+                GRPC_TRANSPORT_SECURITY_TYPE_PROPERTY_NAME, &buffer),
+            GRPC_ALTS_TRANSPORT_SECURITY_TYPE);
+}
+
+TEST_F(DualCredentialsTest, UseTlsCredentials) {
+  ExecCtx exec_ctx;
+  grpc_call_credentials::GetRequestMetadataArgs get_request_metadata_args = {
+      nullptr,
+      CreateAuthContextWithSecurityType(GRPC_TLS_TRANSPORT_SECURITY_TYPE)};
+
+  RunRequestMetadataTest(get_request_metadata_args);
+
+  std::string buffer;
+  EXPECT_EQ(expected_md_.GetStringValue(
+                GRPC_TRANSPORT_SECURITY_TYPE_PROPERTY_NAME, &buffer),
+            GRPC_TLS_TRANSPORT_SECURITY_TYPE);
+}
+
+TEST_F(DualCredentialsTest, NoAuthContext) {
```

Comment:
I don't think this is a realistic use-case.  The auth context should never be null.